hikari/impl/rate_limits.py

```diff
@@ -308,7 +307,7 @@ def __init__(self, name: str, period: float, limit: int) -> None:
         self.limit = limit
         self.period = period
 
-    def acquire(self) -> asyncio.Future[typing.Any]:
+    def acquire(self) -> asyncio.Future[None]:
```

Comment:
random point but the [docs](https://docs.python.org/3/library/asyncio-future.html#asyncio.Future) for asyncio recommend not exposing Future objects in user-facing APIs.
> The rule of thumb is to never expose Future objects in user-facing APIs

Reply:
Ig. How else would you implement this?